runtime/runtime-params-estimator/src/action_costs.rs

```diff
@@ -0,0 +1,626 @@
+            predecessor_id,
+            receiver_id,
+            receipt_id: CryptoHash::new(),
+            receipt: near_primitives::receipt::ReceiptEnum::Action(action_receipt),
+        };
+        let clock = GasCost::measure(self.metric);
+        let outcome = testbed.apply_action_receipt(&receipt);
+        let gas = clock.elapsed();
+        match outcome.status {
+            ExecutionStatus::Unknown => panic!("receipt not applied"),
+            ExecutionStatus::Failure(err) => panic!("failed apply, {err:?}"),
+            ExecutionStatus::SuccessValue(_) | ExecutionStatus::SuccessReceiptId(_) => (),
+        }
+        gas
+    }
+
+    /// Take a function that executes a list of actions on a testbed, execute
+    /// and measure it multiple times and return the average cost.
+    #[track_caller]
+    fn estimate_average_cost(
+        &self,
+        testbed: &mut Testbed,
+        estimated_fn: fn(&Self, &mut Testbed, Vec<Action>) -> GasCost,
+    ) -> GasCost {
+        let num_total_actions = self.actions.len() * self.inner_iters;
+        let actions: Vec<Action> =
+            self.actions.iter().cloned().cycle().take(num_total_actions).collect();
+
+        let gas_results = iter::repeat_with(|| estimated_fn(self, testbed, actions.clone()))
+            .skip(self.warmup)
+            .take(self.outer_iters)
+            .collect();
+
+        // This could be cached for efficiency. But experience so far shows that
+        // reusing caches values for many future estimations leads to the
+        // problem that a single "HIGH-VARIANCE" uncertain estimation can spoil
+        // all following estimations. In this case, rerunning is cheap and it
+        // ensures the base is computed in a very similar state of the machine as
+        // the measurement it is subtracted from.
+        let base =
+            if self.subtract_base { estimated_fn(self, testbed, vec![]) } else { GasCost::zero() };
+
+        let cost_per_tx = average_cost(gas_results);
+        (cost_per_tx - base) / self.inner_iters as u64
+    }
+}
+
+pub(crate) fn create_account_send_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(create_account_action())
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SubOfSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn create_account_send_not_sir(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(create_account_action())
+        .receiver(AccountRequirement::SubOfSigner)
+        .verify_cost(&mut ctx.testbed())
+}
+
+pub(crate) fn create_account_exec(ctx: &mut EstimatorContext) -> GasCost {
+    ActionEstimation::new(ctx)
+        .add_action(create_account_action())
+        .predecessor(AccountRequirement::SameAsSigner)
+        .receiver(AccountRequirement::SubOfSigner)
+        .inner_iters(1) // creating account works only once in a receipt
```

Comment:
Inner iterations generate more actions per receipt. Actions inside a receipt must share the same receiver, because that's defined on the receipt and not on the action. That's why outer iterations (execute multiple transactions) works fine but not inner iterations (execute multiple actions).

Reply:
I guess the easiest way to show this is by looking at the definition of the action:
```
pub struct CreateAccountAction {}
```
It literally has no args, no account name and no keys. A create account action just creates the account for the receiver, and then executes the following actions on behalf of that account. The following actions don't even require access key checks. Usually create account is therefore paired with an add key action, which bootstraps access keys to the account. Otherwise, the account will be locked immediately.